### PR TITLE
Get the Scrapy Cloud job ID from the SHUB_JOBKEY environment variable

### PR DESCRIPTION
**Pull request by Gallaecio**
Based on https://shub.readthedocs.io/en/stable/custom-images-contract.html#shub-jobkey.

Verified manually on a Scrapy Cloud project.

### CODE REVIEW

File: scrapy_zyte_api/_params.py [hunk 1]
```diff
@@ -569,7 +570,7 @@ def __init__(self, crawler, cookies_enabled=None):
         self._automap_params = _load_default_params(settings, "ZYTE_API_AUTOMAP_PARAMS")
         self._browser_headers = _load_browser_headers(settings)
         self._default_params = _load_default_params(settings, "ZYTE_API_DEFAULT_PARAMS")
-        self._job_id = settings.get("JOB")
+        self._job_id = environ.get("SHUB_JOBKEY", None)
```
Comment (kmike) [inline scrapy_zyte_api/_params.py:573]: Is it the case that JOB doesn't work, or is it a clean up to use a documented env variable?

Comment (Gallaecio) [reply scrapy_zyte_api/_params.py:573]: The JOB setting does not work. I am not sure it ever did.

**Review by kmike — COMMENTED** [review]
Thanks @Gallaecio! +1 to merge after fixing tests

File: scrapy_zyte_api/_params.py (hunk 1 above)
Comment (pawelmhm) [reply scrapy_zyte_api/_params.py:573]: I've seen job IDs in several log entries, I didn't investigate how they found their ways to logs, were projects using this plugin or something else. I know about some projects where job key is actually set in settings. 

It is also possible some users went ahead and added JOB key in settings to track it. Perhaps we could keep this for backward compatibility?

Comment (Gallaecio) [reply scrapy_zyte_api/_params.py:573]: > Perhaps we could keep this old way for backward compatibility? It is possible some users are using this plugin outside scrapy cloud and perhaps they rely on this key for something

I would rather mention the backward incompatibility in the changelog, and indicate that if they want something like that they can set the environment variable locally or use `echoData` instead.

**Pull request merged**